vault/audited_headers_test.go

```diff
@@ -181,20 +188,40 @@ func TestAuditedHeadersConfig_ApplyConfig(t *testing.T) {
 		"Content-Type":   {"json"},
 	}
 
-	hashFunc := func(ctx context.Context, s string) (string, error) { return "hashed", nil }
+	salter := &TestSalter{}
 
-	result, err := conf.ApplyConfig(context.Background(), reqHeaders, hashFunc)
+	result, err := conf.ApplyConfig(context.Background(), reqHeaders, salter)
 	if err != nil {
 		t.Fatal(err)
 	}
 
 	expected := map[string][]string{
 		"x-test-header":  {"foo"},
-		"x-vault-header": {"hashed", "hashed"},
+		"x-vault-header": {"hmac-sha256:", "hmac-sha256:"},
+	}
+
+	if len(expected) != len(result) {
+		t.Fatalf("Expected headers count did not match actual count: Expected count %d\n Got %d\n", len(expected), len(result))
 	}
 
-	if !reflect.DeepEqual(result, expected) {
-		t.Fatalf("Expected headers did not match actual: Expected %#v\n Got %#v\n", expected, result)
+	for resultKey, resultValues := range result {
```

Comment:
❓ would we be able to pull out some decent test cases to exercise the code in lots of ways? The pre-existing test just seems to be a happy path test and it would be good to be confident that we've tried to break things and seen it happen in ways we expect.

Reply:
I can add test functions for the case where there's no request headers and where an error is returned by the HashString function.